app/src/main/java/ch/epfl/culturequest/social/Post.java

```diff
@@ -1,40 +1,37 @@
 package ch.epfl.culturequest.social;
 
+import androidx.annotation.NonNull;
+
 import java.util.ArrayList;
-import java.util.Date;
 import java.util.List;
 import java.util.UUID;
-import java.util.concurrent.CompletableFuture;
-import java.util.concurrent.Future;
-import java.util.concurrent.atomic.AtomicBoolean;
-
-import ch.epfl.culturequest.database.Database;
 
 public final class Post {
     private String postId;
     private String uid;
     private String imageUrl;
     private String artworkName;
-    private Date date;
+    private long time;
```

Comment:
Are you sure if we need at some point the date that Date won't be easier to handle, with the day, hour... already accessible?

Reply:
Date format of Java is not compatible with Firebase, it stores it incorrectly, you can check yourself if you want and you will see that the year for example is stored as "123".
So the trick I found is to store a timestamp instead and if you need the date on the app you just need to do "New Date(time)" locally and you will have it ;)

Reply:
Okay then, let's go with it